chirp/drivers/icr6.py

```diff
@@ -0,0 +1,794 @@
+  u8 unknown30;  // 6bee
+  u8 unknown31;  // 6bef
+  u8 unknown32;  // 6bf0
+  u8 unknown33;  // 6bf1
+  u8 civ_address;         // 6bf2: CI-V address, full byte
+  u8 unknown35_6bf3:5,
+     civ_baud_rate:3;     // 6bf3: Index to CIV_BAUD_RATES, range 0-5
+  u8 unknown35_6bf4:7,
+    civ_transceive:1;    // 6bf4: Report frequency and mode changes
+  u8 unknown37;  // 6bf5
+  u8 unknown38;  // 6bf6
+  u8 unknown39;  // 6bf7
+  u8 unknown40;  // 6bf8
+  u8 unknown41;  // 6bf9
+  u8 unknown42;  // 6bfa
+  u8 unknown43;  // 6bfb
+  u8 unknown44;  // 6bfc
+  u8 unknown45;  // 6bfd
+  u8 unknown46;  // 6bfe
+  u8 unknown47;  // 6bff
+  u8 unknown48;  // 6c00
+  u8 unknown49;  // 6c01
+  u8 unknown50;  // c602
+  u8 unknown51;  // c603
+  u8 unknown52h_6c04:3,   // Fixed 001 seen during tests
+     dial_function:1,     // 0=Tuning Dial, 1=Audio Volume
+     unknown52m_6c04:2,   // Fixed 10 seen during tests
+     mem_display_type:2;  // 00=Freq, 01=BankName, 02=MemName, 03=ChNum
+  u8 unknown54;  // 6c05
+  u8 unknown55;  // 6c06
+  u8 unknown56;  // 6c07
+  u8 unknown57;  // 6c08
+  u8 unknown58;  // 6c09
+  u8 unknown59;  // 6c0a
+  u8 unknown60;  // 6c0b
+  u8 unknown61;  // 6c0c
+  u8 unknown62;  // 6c0d
+  u8 unknown63;  // 6c0e
+  u8 unknown53;  // 6c0f
+} settings;
+
+
+// Unknown. Coded as 15 blocks of 16-bytes. 0x6C10 to 0x6CFF
+struct {
+  u8 unknown0;
+  u8 unknown1;
+  u8 unknown2;
+  u8 unknown3;
+  u8 unknown4;
+  u8 unknown5;
+  u8 unknown6;
+  u8 unknown7;
+  u8 unknown8;
+  u8 unknown9;
+  u8 unknown10;
+  u8 unknown11;
+  u8 unknown12;
+  u8 unknown13;
+  u8 unknown14;
+  u8 unknown15;
+} mystery4[15];
+
+// Device comment string. Grab it from the ICF?
+struct { // Start: 6D00, End: 6D0F
+  char comment[16];
+} device_comment;
+
+// 22x ASCII-coded bank names
+// 0x6D10 to 0x6DBF inclusive
+struct {
+  char name[6];
+  u8 unknown0; // Padding?
+  u8 unknown1; // Padding?
+} bank_names[22];
+
+// ASCII-coded scan link names x10: 0x6DC0 to 0x6E0F
+struct {
+  char name[6];
+  u8 unknown0;
+  u8 unknown1;
+} prog_scan_link_names[10];
+
+// Unknown: Tail end of memory? 6E10 onwards
+struct {  // Start: 6E10
+  u8 unknown0;
+} mystery5[64];
+
+struct {
+  // The string "IcomCloneFormat3" at end of block
+  char footer[16];
+} footer;
+"""
+
+TONE_MODES = ["", "TSQL", "TSQL-R", "DTCS", "DTCS-R"]
+#  Sub-audible tone settings:
+#    "TSQL" (with pocket beep), "TSQL" (no beep),
+#    "DTCS" (with pocket beep), "DTCS" (no beep),
+#    "TQQL-R" (Tone Squelch Reverse),
+#    "DTCS-R" (DTCS Reverse)”,
+#    "OFF"
+TONES = (67.0,  69.3,  71.9,  74.4,  77.0,  79.7,  82.5,  85.4,
+         88.5,  91.5,  94.8,  97.4, 100.0, 103.5, 107.2, 110.9,
+         114.8, 118.8, 123.0, 127.3, 131.8, 136.5, 141.3, 146.2,
+         151.4, 156.7, 159.8, 162.2, 165.5, 167.9, 171.3, 173.8,
+         177.3, 179.9, 183.5, 186.2, 189.9, 192.8, 196.6, 199.5,
+         203.5, 206.5, 210.7, 218.1, 225.7, 229.1, 233.6, 241.8,
+         250.3, 254.1)
```

Comment:
Is this different than one of the sets in chirp_common? If so, it'd be nice to copy and modify it here so it's easier to tell what the delta is. Like, if it's missing 100.0 or something, then:
```
TONES = list(chirp_common.TONES)
TONES.remove(100.0)
```

Reply:
Fixed... same list.